### ISSUE TITLE
npm audit security report

### ISSUE THREAD

**Issue by MarcLoupias** [jsonresume/resume-cli#289]
Hello, i would like to use the nice jsonresume ecosystem for my personal resume but there is a lot of security issues with it.

Running `npm audit` with the `master` branch (using `master` branch is suggested in the #288 issue) output this : 

```
                       === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Cross-Site Scripting                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ mustache                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.2.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > jsonresume-theme-crisp > mustache                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/62                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Cross-Site Scripting                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ mustache                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.2.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > jsonresume-theme-modern > mustache                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/62                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Cross-Site Scripting                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ mustache                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.2.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > resume-to-html > mustache                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/62                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Cross-Site Scripting                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ mustache                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.2.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > resume-to-html > resume-to-markdown > mustache             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/62                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Cross-Site Scripting                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ mustache                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.2.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > resume-to-text > mustache                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/62                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Cross-Site Scripting                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ mustache                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.2.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > resume-to-text > resume-to-html > mustache                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/62                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Cross-Site Scripting                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ mustache                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.2.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > resume-to-text > resume-to-html > resume-to-markdown >     │
│               │ mustache                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/62                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Critical      │ Command Injection                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ open                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > open                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/663                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Cross-Site Scripting                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ handlebars                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.0.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > jsonresume-theme-flat > handlebars                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/61                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ lodash                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.17.5                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > jsonresume-theme-crisp > lodash                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/577                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ lodash                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.17.5                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > jsonresume-theme-modern > lodash                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/577                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ lodash                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.17.5                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > resume-to-html > lodash                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/577                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ lodash                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.17.5                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > resume-to-text > resume-to-html > lodash                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/577                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Incorrect Handling of Non-Boolean Comparisons During         │
│               │ Minification                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ uglify-js                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >= 2.4.24                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > jsonresume-theme-flat > handlebars > uglify-js             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/39                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ uglify-js                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.6.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > jsonresume-theme-flat > handlebars > uglify-js             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/48                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Large gzip Denial of Service                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ superagent                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.7.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ [dev]                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 863817fae08b0291cb7ccc8ab600bc70770402047a0b64141ecf2b0d6db… │
│               │ > superagent                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/479                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 16 vulnerabilities (7 low, 8 high, 1 critical) in 981 scanned packages
  16 vulnerabilities require manual review. See the full report for details.

```

In summary : 

```
jsonresume-theme-crisp > mustache
jsonresume-theme-modern > mustache
resume-to-html > mustache
resume-to-html > resume-to-markdown > mustache
resume-to-text > mustache
resume-to-text > resume-to-html > mustache
resume-to-text > resume-to-html > resume-to-markdown > mustache
open
jsonresume-theme-flat > handlebars
jsonresume-theme-crisp > lodash 
jsonresume-theme-modern > lodash
resume-to-html > lodash
resume-to-text > resume-to-html > lodash
jsonresume-theme-flat > handlebars > uglify-js
jsonresume-theme-flat > handlebars > uglify-js
superagent
```

This is super annoying.

The issue with the [open package](https://www.npmjs.com/package/open) is really problematic as it is critical ([command injection](https://www.npmjs.com/advisories/663)) and there is no patch to fix it !

Could it be possible to remove the open package and fix the rest ?

The [resume-to-html package](https://www.npmjs.com/package/resume-to-html) does not exist anymore in the @thomasdavis repositories list. So impossible to PR.

**Comment by thomasdavis** [comment]
Your PR doesn't fix the `open` bug right so we should look at replacing it?

**Comment by MarcLoupias** [comment]
Yup. There is no fix for `open` package. The package is unmaintained. So the PR replace it with [`opn`](https://github.com/sindresorhus/opn).

**Comment by rbardini** [comment]
Fixed in #369.
